### PR TITLE
Modernize Jakefile + fix project references breaks

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Merges parser, core, and compiler back into compiler per request

Rewrites the Jakefile and fixes a variety of tsbuild bugs.

Includes LKG update; see me offline for an analysis of what's i nthere

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
CI failures are just node6 OOMing as usual

**Comment by weswigham** [comment]
@RyanCavanaugh node6 doesn't usually OOM; it usually just times out, and usually only on travis (as we've enver had the issue on VSTS), AFAIK (and usually not repeatable if you rerun the build). I'd find consistent OOMs (even if confined to one version) fairly concerning; it probably means that on that version we OOM but on the others we're close to being OOM.

**Comment by weswigham** [comment]
For reference, in `node8` the build step that runs OOM seems to take 85s on travis, while on `node6`, it runs OOM after just 97s (and it'll take longer when nearing OOM as it frantically GCs). The similarity would lead me to believe that, barring running into some memory blowup issue confined to node6 late in the build, that that build is on the knife's edge of even being possible.

@rbuckton can probably relate, considering he's already had to acknowledge that running all the builds in one processes like that isn't possible when running under `gulp`, so it's not too surprising `tsbuild` would hit the same limitations...

**Comment by RyanCavanaugh** [comment]
K, investigating the memory thing

File: src/compiler/tsbuild.ts [hunk 1]
```diff
@@ -1095,16 +1109,18 @@ namespace ts {
             return resolvedNames;
         }
 
-        function buildAllProjects() {
+        function buildAllProjects(): number {
```
Comment (sandersn) [inline src/compiler/tsbuild.ts]: why isn't this return type `ExitStatus`?

File: src/tsserverLibrary/tsconfig.json [hunk 2]
```diff
@@ -1,19 +0,0 @@
-{
-    "extends": "../tsconfig-base",
```
Comment (sandersn) [inline src/tsserverLibrary/tsconfig.json]: what are these files replaced by (if anything)?

File: Jakefile.js [hunk 3]
```diff
@@ -921,92 +505,123 @@ function runConsoleTests(defaultReporter, runInParallel) {
...
+    if (fs.existsSync(Paths.baselines.localRwc)) {
+        del.sync(Paths.baselines.localRwc);
+    }
 
-// Browser tests
-var nodeServerOutFile = "tests/webTestServer.js";
-var nodeServerInFile = "tests/webTestServer.ts";
-compileFile(nodeServerOutFile, [nodeServerInFile], [builtLocalDirectory, tscFile], [], /*useBuiltCompiler:*/ true, { noOutFile: true, lib: "es6" });
+    jake.mkdirP(Paths.baselines.local);
+    jake.mkdirP(Paths.baselines.localRwc);
```
Comment (weswigham) [inline Jakefile.js]: This is going to make an `internal` folder for people without `internal` checked out. We probably don't want that.

File: src/tsserverLibrary/tsconfig.json (hunk 2 above)
Comment (RyanCavanaugh) [reply src/tsserverLibrary/tsconfig.json]: Nothing

**Comment by rbuckton** [comment]
Apparently we are no longer writing out _typescript.js_ in the jake builds.

**Pull request merged**